src/IRCNetwork.cpp

```diff
@@ -942,6 +942,35 @@ bool CIRCNetwork::DelChan(const CString& sName) {
     return false;
 }
 
+bool CIRCNetwork::MoveChan(const CString& sName, unsigned int uIndex) {
```

Comment:
Every time this function is called, you're checking first whether the channel exists.

Maybe pass the `CChannel*` instead, to avoid looking it up again? Same for swap

Another option is to return a better error message from here instead of just "false", then you won't need to check it before calling, but just need to show the error message returned from here

Reply:
Passing the `CChan*` would be cleaner. This and removing the redundant check on `uIndex` would make both `MoveChan` and `SwapChan` void functions and leave all input validation at the call sites.

Reply:
I've just realized that `CChan*` wouldn't be enough, because it will still require searching for it in the vector.

There are other functions here already which return the error via `CString& sError`